tools/buildHeaders/jsonToSpirv.cpp

```diff
@@ -471,6 +471,36 @@ unsigned int NumberStringToBit(const std::string& str)
     return bit;
 }
 
+// Given two pairs (name and in core) compares if the order is correct for naming
+// conventions. The conventions are:
+// * Core
+// * KHR
+// * EXT
+// * Vendor (no preference between vendors)
+//
+// Returns true if the order is valid.
+bool SuffixComparison(const std::string& prev, bool prevCore,
+                      const std::string& cur, bool curCore)
+{
+  if (prev == cur) return true;
```

Comment:
IIUC, this condition is only hit when encountering two instructions with the same name. Shouldn't we error out instead? Do we reject duplicates somewhere else?

Reply:
We didn't reject duplicates previously, so I made that change.

Reply:
Good to do IMO, it buys us an extra check for very little work. Thanks!